applications/solvers/additiveFoam/movingHeatSource/refinementControllers/uniformIntervals/uniformIntervals.C

```diff
@@ -49,153 +54,148 @@ Foam::refinementControllers::uniformIntervals::uniformIntervals
-    boundingBox_ /= 2.0;
+    // TODO: interval time and update time need to be relative to start time
+
+    // set end time for AMR update to minimum of solution time and max beam time
+    forAll(sources_, i)
+    {
+        endTime_ = max(sources_[i].beam().endTime(), endTime_);
+    }
+
+    endTime_ = min(endTime_, mesh.time().endTime().value());
+
+    Info << "endTime: " << endTime_ << endl;
+
+    // set interval time to the number of intervals between start and end
+    intervalTime_ =
+        max(0, endTime_ - mesh.time().startTime().value()) / intervals_;
+
+    Info<< "AMR time interval: " << intervalTime_ << " s" << endl;
+
+    Info<< "refinement bounding box: " << boundingBox_ << endl;
 }
 
 
 // * * * * * * * * * * * * * * Member Functions  * * * * * * * * * * * * * * //
 
 bool Foam::refinementControllers::uniformIntervals::update()
 {
-    //- Update refinement field at update time
-    if ((refinementController::update())
-        &&
-        (mesh_.time().value() >= updateTime_))
+    if ((updateTime_ - mesh_.time().value()) < small)
     {
-        //- Update refinement index
+        // update refinement index and next refinement time
         lastRefinementIndex_ = mesh_.time().timeIndex();
-        
-        //- Call an extra mesh refinement on the first time
-        if (mesh_.time().timeIndex() == 0)
+        updateTime_ = mesh_.time().value() + intervalTime_;
+
+        //- Set initial refinement field in base class
+        refinementController::setRefinementField();
+
+        if ((endTime_ - mesh_.time().value()) < small)
         {
-            lastRefinementIndex_ = 1;
+            Info << "continuing AMR checks for possible mesh coarsening" << endl;
+            return true;
         }
-        
-        //- Initialize refinement marker field in base class
-        refinementController::initializeRefinementField();
-    
-        //- Update next refinement time
-        const scalar currTime = mesh_.time().value();
-        updateTime_ = currTime + intervalSize_;
-        
-        //- Buffer for bounding box calculations
-        const vector extend = 1e-10 * vector::one;
-        
+
+        //- Calculate the bounding box for each cell
+        List<treeBoundBox> cellBbs(mesh_.nCells());    
         const pointField& points = mesh_.points();
-        
-        //- Update refinement marker field
+        const vector extend = 1e-10 * vector::one;
+
+        forAll(mesh_.cells(), celli)
+        {
+            treeBoundBox cellBb(point::max, point::min);
+
+            const labelList& vertices = mesh_.cellPoints()[celli];
+
+            forAll(vertices, j)
+            {
+                cellBb.min()
+                    = min(cellBb.min(), points[vertices[j]] - extend);
+                cellBb.max()
+                    = max(cellBb.max(), points[vertices[j]] + extend);
+            }
+
+            cellBbs[celli] = cellBb;
+        }
+
+        //- Update refinement marker field:
```

Comment:
can we use a consistent comment format? I've been using:

```
//- comment line 1
//  comment line 2
//  blah blah blah
```

sometimes you use `//-`, sometimes just `//`. i don't particularly care which we settle one, just that it's consistent

Reply:
Agreed. I tried to be more consistent